code/modules/mining/lavaland/necropolis_chests.dm

```diff
@@ -438,4 +439,183 @@
+
+	force = Clamp((ghost_counter * 4), 0, 75)
+	user.visible_message("<span class='danger'>[user] strikes with the force of [ghost_counter] vengeful spirits!</span>")
+	..()
+
+/obj/item/weapon/melee/ghost_sword/hit_reaction(mob/living/carbon/human/owner, attack_text, final_block_chance, damage, attack_type)
+	var/ghost_counter = ghost_check()
+	final_block_chance += Clamp((ghost_counter * 5), 0, 75)
+	owner.visible_message("<span class='danger'>[owner] is protected by a ring of [ghost_counter] ghosts!</span>")
+	return ..()
+//Blood
+
+/obj/item/weapon/dragons_blood
+	name = "bottle of dragons blood"
+	desc = "You're not actually going to drink this, are you?"
+	icon = 'icons/obj/wizard.dmi'
+	icon_state = "vial"
+
+/obj/item/weapon/dragons_blood/attack_self(mob/living/carbon/human/user)
+	if(!istype(user))
+		return
+
+	var/mob/living/carbon/human/H = user
+	var/random = rand(1,3)
+
+	switch(random)
+		if(1)
+			user << "<span class='danger'>Other than tasting terrible, nothing really happens.</span>"
+		if(2)
+			user << "<span class='danger'>Your flesh begins to melt! Miraculously, you seem fine otherwise.</span>"
+			H.set_species(/datum/species/skeleton)
+		if(3)
+			user << "<span class='danger'>You don't feel so good...</span>"
+			message_admins("[key_name_admin(user)](<A HREF='?_src_=holder;adminplayerobservefollow=\ref[user]'>FLW</A>) has started transforming into a dragon via dragon's blood.")
+			H.ForceContractDisease(new /datum/disease/transformation/dragon(0))
+
+	playsound(user.loc,'sound/items/drink.ogg', rand(10,50), 1)
+	qdel(src)
+
+/datum/disease/transformation/dragon
+	name = "dragon transformation"
+	cure_text = "nothing"
+	cures = list("adminordrazine")
+	agent = "dragon's blood"
+	desc = "What do dragons have to do with Space Station 13?"
+	stage_prob = 20
+	severity = BIOHAZARD
+	visibility_flags = 0
+	stage1	= list("Your bones ache.")
+	stage2	= list("Your skin feels scaley.")
+	stage3	= list("<span class='danger'>You have an overwhelming urge to terrorize some peasants.</span>", "<span class='danger'>Your teeth feel sharper.</span>")
+	stage4	= list("<span class='danger'>Your blood burns.</span>")
+	stage5	= list("<span class='danger'>You're a fucking dragon.</span>")
+	new_form = /mob/living/simple_animal/hostile/megafauna/dragon/lesser
+
+
+//Lava Staff
+
+/obj/item/weapon/lava_staff
+	name = "staff of lava"
+	desc = "The ability to fill the emergency shuttle with lava. What more could you want out of life?"
+	icon_state = "staffofstorms"
+	item_state = "staffofstorms"
+	icon = 'icons/obj/guns/magic.dmi'
+	slot_flags = SLOT_BACK
+	item_state = "staffofstorms"
+	w_class = 4
+	force = 25
+	damtype = BURN
+	burn_state = LAVA_PROOF
+	hitsound = 'sound/weapons/sear.ogg'
+	var/turf_type = /turf/open/floor/plating/lava/smooth
+	var/cooldown = 200
+	var/timer = 0
+	var/banned_turfs
+
+/obj/item/weapon/lava_staff/New()
+	. = ..()
+	banned_turfs = typecacheof(list(/turf/open/space/transit, /turf/closed))
```

Comment:
Lava staffing space is actually quite common. Unless Kor says otherwise, it'll stay in, silly as it is.